mailmerge/api.py

```diff
@@ -94,6 +91,39 @@ def addattachments(message, template_path):
         original_text = message.get_payload()
         multipart_message.attach(email.mime.text.MIMEText(original_text))
         message = multipart_message
+    # HACK: For Python2 (see comments in `_create_boundary`)
+    message = _create_boundary(message)
+    return message
+
+
+def convert_markdown(message):
+    """Convert markdown in message text to HTML."""
+    if not message['Content-Type'].startswith("text/markdown"):
+        return message
+    del message['Content-Type']
+    # Convert the text from markdown and then make the message multipart
+    message = make_message_multipart(message)
+    for payload_item in set(message.get_payload()):
+        # Assume the plaintext item is formatted with markdown.
+        # Add corresponding HTML version of the item as the last part of
+        # the multipart message (as per RFC 2046)
+        if payload_item['Content-Type'].startswith('text/plain'):
+            original_text = payload_item.get_payload()
+            html_text = markdown.markdown(original_text)
+            html_payload = future.backports.email.mime.text.MIMEText(
+                "<html><body>{}</body></html>".format(html_text),
+                "html",
+            )
+            message.attach(html_payload)
+    return message
```

Comment:
Should this function delete the custom header?

Reply:
Yes, it should. L103 deletes the header before converting the message to multipart.

Reply:
Whoops, I missed that.